### ISSUE TITLE
Simplify Eq simplification

### ISSUE THREAD

**Issue by nfrisby** [nfrisby/frags#36]
`Eq 'Nil` should decompose like `Set`.

```haskell
Eq :: Frag k -> Frag k -> Constraint

Eq (frL :+- a) frR   ~   Eq frL (fr :-+ a)
Eq 'Nil 'Nil   ~   ()
Eq 'Nil (fr :+- a)   ~   ( Eq 'Nil (FragNE a fr) , Eq 'Nil (FragEQ a fr :+- '()) )   when k /~ ()
```

**Comment by nfrisby** [comment]
I think this isolates deqs and dneqs to interactions between from `Set` or `Eq 'Nil` applied to `FragNE` and/or `FragEQ`.

**Comment by nfrisby** [comment]
Closed by `862074094040edce44df6a0bdc2ec0216829c475`, which changes the docs instead of changing the implementation. Thus more of a "WontFix" resolution instead of "Done".

Once I started to add the `IsNil :: Frag k -> ()` type family to the plugin's simplification logic, I realized it would interfere with the inert frag substitution. See the updated `README.md` for details. Ultimate takeaway: 

  * `[FOR-DISCUSSION-EqFrag-Nil-tally]` seems mutually exclusive with the inert frag substitution

  * Without `[FOR-DISCUSSION-EqFrag-Nil-tally]`, the improvement rules `[EqFrag-Nil-FragEQ-Nil-improve-pos]` and `[EqFrag-Nil-FragEQ-Nil-improve-neg]` do not get activated by `Nil`-`Nil` `EqFrag` constraints.

  * So we add `[EqFrag-Nil-Nil-improve-pos]` and `[EqFrag-Nil-Nil-improve-neg]` in order to recover the corresponding improvements in the absence of `[FOR-DISCUSSION-EqFrag-Nil-tally]`.
